src/registrar/models/domain.py

```diff
@@ -91,221 +93,193 @@ class Status(models.TextChoices):
-        return domain
+        # previously existed but has been deleted from the registry
+        DELETED = "deleted"
+
+        # the state is indeterminate
+        UNKNOWN = "unknown"
 
     @classmethod
     def available(cls, domain: str) -> bool:
-        """Check if a domain is available.
+        """Check if a domain is available."""
+        if not cls.string_could_be_domain(domain):
+            raise ValueError("Not a valid domain: %s" % str(domain))
+        req = commands.CheckDomain([domain])
+        return registry.send(req).res_data[0].avail
 
-        Not implemented. Returns a dummy value for testing."""
-        return False  # domain_check(domain)
+    @classmethod
+    def registered(cls, domain: str) -> bool:
+        """Check if a domain is _not_ available."""
+        return not cls.available(domain)
 
-    def transfer(self):
-        """Going somewhere. Not implemented."""
-        pass
+    @property
+    def contacts(self) -> dict[str, str]:
+        """
+        Get a dictionary of registry IDs for the contacts for this domain.
 
-    def renew(self):
-        """Time to renew. Not implemented."""
-        pass
+        IDs are provided as strings, e.g.
 
-    def _get_property(self, property):
-        """Get some info about a domain."""
-        if not self.is_active:
-            return None
-        if not hasattr(self, "info"):
-            try:
-                # get info from registry
-                self.info = {}  # domain_info(self.name)
-            except Exception as e:
-                logger.error(e)
-                # TODO: back off error handling
-                return None
-        if hasattr(self, "info"):
-            if property in self.info:
-                return self.info[property]
-            else:
-                raise KeyError(
-                    "Requested key %s was not found in registry data." % str(property)
-                )
-        else:
-            # TODO: return an error if registry cannot be contacted
-            return None
-
-    @transition(field="is_active", source="*", target=True)
-    def activate(self):
-        """This domain should be made live."""
-        DomainApplication = apps.get_model("registrar.DomainApplication")
-        if hasattr(self, "domain_application"):
-            if self.domain_application.status != DomainApplication.APPROVED:
-                raise ValueError("Cannot activate. Application must be approved.")
-        if Domain.objects.filter(name=self.name, is_active=True).exists():
-            raise ValueError("Cannot activate. Domain name is already in use.")
-        # TODO: depending on the details of our registry integration
-        # we will either contact the registry and deploy the domain
-        # in this function OR we will verify that it has already been
-        # activated and reject this state transition if it has not
-        pass
+            {"registrant": "jd1234", "admin": "sh8013",...}
```

Comment:
**comment (non-blocking)**: In the future, when implementing this method, I think that we now have these dictionary keys "registrant", "admin", etc. as class constants under `PublicContact`, maybe we should re-use those comments here.